flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/MasterTriggerRestoreHook.java

```diff
@@ -95,14 +95,17 @@ default void close() throws Exception {
 	 * the checkpoint metadata under the hooks identifier (see {@link #getIdentifier()}).
 	 *
 	 * <p>If the action by this hook needs to be executed synchronously, then this method should
-	 * directly execute the action synchronously and block until it is complete. The returned future
-	 * (if any) would typically be a completed future.
+	 * directly execute the action synchronously. The returned future (if any) would typically be a
+	 * completed future.
 	 *
 	 * <p>If the action should be executed asynchronously and only needs to complete before the
 	 * checkpoint is considered completed, then the method may use the given executor to execute the
 	 * actual action and would signal its completion by completing the future. For hooks that do not
 	 * need to store data, the future would be completed with null.
 	 *
+	 * <p>Please note that this method should be non-blocking. Any heavy operation like IO operation
+	 * should be executed asynchronously with given executor.
```

Comment:
This will require a mention in the release notes, as previously we were encouraging the calls to be blocking.

Can you add an appropriate reference in the jira ticket? (there is a filed for release notes)

Reply:
Nice suggestion!